feat: rules refactoring / add separate arazzo rule-set (#1800)
Redocly/redocly-cli: .changeset/strong-ties-argue.md

---
"@redocly/openapi-core": patch

> "@redocly/cli": patch

Author: - change to 'minor'

---

> Introduced `struct` rule. Added Arazzo version 1 ruleset.

Reviewer: ```suggestion
Introduced `struct` rule.
Added Arazzo version 1 ruleset.
```

> Introduced the `struct` rule and deprecated the `spec` rule. Added the `spec` ruleset, which enforces compliance with the specifications.

Reviewer: ```suggestion
Introduced the `struct` rule and deprecated the `spec` rule.
Added the `spec` ruleset, which enforces compliance with the specifications.
```

---
"@redocly/openapi-core": patch
"@redocly/cli": patch
---

> Introduced `struct` rule.
> Added Arazzo version 1 ruleset.

Reviewer 2: ```suggestion
Introduced `struct` rule. Added a `spec` ruleset which enforces compliance with the specifications.
```
